### ISSUE TITLE
[BUG]: Installation error on debian 12 and ubuntu lts 22.04.1

### ISSUE THREAD

**Issue by brunobelorte** [rapidsai/cugraph#4265]
### Version

11

### Which installation method(s) does this occur on?

Conda, Pip

### Describe the bug.

When trying to install the package using pip and conda, like [ readme.md](https://github.com/rapidsai/cugraph/blob/branch-24.06/python/nx-cugraph/README.md)  I got errors related to package parameters

### Relevant log output

```shell
`(base) bruno@bruno:~$ conda install -c rapidsai -c conda-forge -c nvidia nx-cugraph
Channels:
 - rapidsai
 - conda-forge
 - nvidia
 - defaults
Platform: linux-64
Collecting package metadata (repodata.json): done
Solving environment: / warning  libmamba Added empty dependency for problem type SOLVER_RULE_UPDATE
failed

LibMambaUnsatisfiableError: Encountered problems while solving:
  - package nx-cugraph-23.10.00-py310_0 requires python >=3.10,<3.11.0a0, but none of the providers can be installed

Could not solve for environment specs
The following packages are incompatible
├─ nx-cugraph is installable with the potential options
│  ├─ nx-cugraph [23.10.00|23.12.00|24.02.00] would require
│  │  └─ python >=3.10,<3.11.0a0 , which can be installed;
│  └─ nx-cugraph [23.10.00|23.12.00|24.02.00] would require
│     └─ python >=3.9,<3.10.0a0 , which can be installed;
└─ pin-1 is not installable because it requires
   └─ python 3.11.* , which conflicts with any installable versions previously reported.


bruno@bruno:~$ pip install nx-cugraph-cu11 --extra-index-url https://pypi.nvidia.com
Defaulting to user installation because normal site-packages is not writeable
Looking in indexes: https://pypi.org/simple, https://pypi.nvidia.com
Collecting nx-cugraph-cu11
  Using cached https://pypi.nvidia.com/nx-cugraph-cu11/nx_cugraph_cu11-24.2.0-py3-none-any.whl (117 kB)
Collecting cupy-cuda11x>=12.0.0 (from nx-cugraph-cu11)
  Using cached cupy_cuda11x-13.0.0-cp311-cp311-manylinux2014_x86_64.whl.metadata (2.6 kB)
Requirement already satisfied: networkx>=3.0 in ./.local/lib/python3.11/site-packages (from nx-cugraph-cu11) (3.2.1)
Requirement already satisfied: numpy>=1.21 in ./.local/lib/python3.11/site-packages (from nx-cugraph-cu11) (1.26.4)
Collecting pylibcugraph-cu11==24.2.* (from nx-cugraph-cu11)
  Using cached pylibcugraph-cu11-24.2.0.tar.gz (6.9 kB)
  Preparing metadata (setup.py) ... error
  error: subprocess-exited-with-error
  
  × python setup.py egg_info did not run successfully.
  │ exit code: 1
  ╰─> [6 lines of output]
      Traceback (most recent call last):
        File "<string>", line 2, in <module>
        File "<pip-setuptools-caller>", line 34, in <module>
        File "/tmp/pip-install-cdx23gwm/pylibcugraph-cu11_7f04b91adaf944e89ef2bd9b79bd1bca/setup.py", line 90, in <module>
          raise RuntimeError("Bad params")
      RuntimeError: Bad params
      [end of output]
  
  note: This error originates from a subprocess, and is likely not a problem with pip.
error: metadata-generation-failed

× Encountered error while generating package metadata.
╰─> See above for output.

note: This is an issue with the package mentioned above, not pip.
hint: See above for details.
`
```

### Environment details

Linux bruno 6.1.0-18-amd64 #1 SMP PREEMPT_DYNAMIC Debian 6.1.76-1 (2024-02-01) x86_64 GNU/Linux
with Python 3.11.2

Linux turing 6.5.0-25-generic #25~22.04.1-Ubuntu SMP PREEMPT_DYNAMIC Tue Feb 20 16:09:15 UTC 2 x86_64 x86_64 x86_64 GNU/Linux
with Python 3.11.5

### Code of Conduct

- [X] I agree to follow cuGraph's Code of Conduct
- [X] I have searched the [open bugs](https://github.com/rapidsai/cugraph/issues?q=is%3Aopen+is%3Aissue+label%3Abug) and have found no duplicates for this bug report

**Comment by zdimension** [comment]
Getting the same on Debian 11 when installing via pip
- Linux domino 6.1.0-0.deb11.17-amd64 #⁠1 SMP PREEMPT_DYNAMIC Debian 6.1.69-1~bpo11+1 (2024-01-05) x86_64 GNU/Linux
- Python 3.10.6

**Comment by brunobelorte** [comment]
I installed this using the rapids tutorial
https://docs.rapids.ai/install

Command

conda create --solver=libmamba -n rapids-24.04 -c rapidsai-nightly -c conda-forge -c nvidia  \
    rapids=24.04 python=3.11 cuda-version=12.0

**Comment by rlratzel** [comment]
Hi @brunobelorte , I noticed you initially had the problem when trying to install `nx-cugraph`, but you mentioned you recently resolved it by installing `rapids=24.04` using the RAPIDS install docs.  

A few things jump out that I'm wondering about: 
* For the conda install you tried, it appears conda is choosing an old version of nx-cugraph (`nx-cugraph-23.10.00-py310_0`).  I wonder if there's some additional constraints somewhere on your system (a `.condarc` somewhere, or perhaps an old CUDA driver) causing conda to pick an old version.  I'm thinking the `rapids-24.04` meta-package has additional specifications which force conda to install other prerequisites that result in a working environment?  Or maybe it's because you just ran it on a later date when a more recent `nx-cugraph` was available?  I'm interested in learning more in case we need to update our conda recipe with additional specifications.
* The `pip install` command appears to have picked a more recent version of `nx-cugraph` (24.2) but ran into a problem installing a dependency (`pylibcugraph`).  The "Bad params" error makes me think an older version of setuptools/pip is being used.
* Can you try installing `nx-cugraph` again now that a newer version is available?  I'm assuming conda should find version `24.04`, and if not, can you try installing by specifying `nx-cugraph=24.04`?

**Comment by kingmesal** [comment]
reopen if this is still an issue.
